### PR TITLE
Add ldap capabilities to authorization checking to allow group author…

### PR DESCRIPTION
**Pull request by chambrid**
…ization management.

### CODE REVIEW

File: src/scripts/ibmcloud-auth.js [hunk 1]
```diff
@@ -111,6 +127,40 @@ if (env.readerUsers) {
 	READER_EMAILS = env.readerUsers.split(',');
 }
 
+function ldapInit() {
+	var promise = new Promise((resolve, reject) => {
+		if (env.ldapServer && env.ldapPort && env.ldapBindUser && env.ldapBindPassword) {
+			var ldapUrl = `${env.ldapProtocol}://${env.ldapServer}:${env.ldapPort}`;
+			POWER_GROUPS = env.ldapPowerGroups.split(';');
```
Comment (kholdaway) [inline src/scripts/ibmcloud-auth.js]: Why semicolon?  The other env uses commas.

 HUBOT_IBMCLOUD_POWERUSERS=<comma-separated list of power-user emails -- no spaces!>
 HUBOT_IBMCLOUD_READERUSERS=<comma-separated list of reader-user emails -- no spaces!>

Comment (kevan-ibm) [inline src/scripts/ibmcloud-auth.js:134]: Why semicolon?  The other env uses commas.
HUBOT_IBMCLOUD_POWERUSERS=comma-separated list of power-user emails -- no spaces!
HUBOT_IBMCLOUD_READERUSERS=comma-separated list of reader-user emails -- no spaces!

Comment (chambrid) [reply src/scripts/ibmcloud-auth.js:134]: LDAP users have commas in the distinguished names: `ou=mathematicians,dc=example,dc=com`

File: src/scripts/ibmcloud-auth.js [hunk 2]
```diff
@@ -111,6 +127,40 @@ if (env.readerUsers) {
...
+				bot.logger.debug(`${TAG}: LDAP READER_GROUPS=${READER_GROUPS}`);
+				bot.logger.info(`${TAG}: Attemping to connect to LDAP at ${ldapUrl}.`);
+			}
+			ldapClient = ldap.createClient({
+				url: ldapUrl,
+				connectTimeout: 5000,
+				timeout: 10000
+			});
+			ldapClient.bind(env.ldapBindUser, env.ldapBindPassword, function(err) {
+				if (err) {
```
Comment (kevan-ibm) [inline src/scripts/ibmcloud-auth.js:147]: Should this reject the promise if it failed to bind?

Comment (chambrid) [reply src/scripts/ibmcloud-auth.js:147]: My intent was to allow authorization initialization even if there was an error with LDAP as I want to make sure the auth middleware and NLC are in place so the bot is still locked down for usage.

**Comment by kevan-ibm** [comment]
Looks good.

**Pull request merged**